src/Chapter1.hs

```diff
@@ -187,462 +181,462 @@ ghci> :q
+  | c == 'e' || c == 'E' = True
+  | c == 'i' || c == 'I' = True
+  | c == 'o' || c == 'O' = True
+  | c == 'u' || c == 'U' = True
+  | otherwise = False
+
+-- |
+-- == Local variables and functions
+--
+-- So far, we've been playing only with simple expressions and function
+-- definitions. However, in some cases, expressions may become complicated, and it
+-- could make sense to introduce some helper variables.
+--
+-- You can use the let-in construction in Haskell to define variables.
+-- Here goes an example:
+--
+-- @
+-- half :: Int -> Int
+-- half n = let halfN = div n 2 in halfN
+-- @
+--
+-- ♫ NOTE: __let-in__ is also an expression! You can't just define variables; you
+--   also need to return some expression that may use defined variables.
+--
+-- The syntax for defining multiple variables requires to care about indentation
+-- more, but there is nothing special in it as well:
+--
+-- @
+-- halfAndTwice :: Int -> (Int, Int)
+-- halfAndTwice n =
+--     let halfN = div n 2
+--         twiceN = n * 2
+--     in (halfN, twiceN)
+-- @
+--
+-- In addition to let-in (or sometimes even alternatively to let-in) you can use
+-- the __where__ construction to define local variables and functions.
+-- And, again, the example:
+--
+-- @
+-- pythagoras :: Double -> Double -> Double
+-- pythagoras a b = square a + square b
+--   where
+--     square :: Double -> Double
+--     square x = x ^ 2
+-- @
+--
+-- You can define multiple functions inside __where__!
+-- Just remember to keep proper indentation.
+
+-- |
+-- =⚔️= Task 9
+--
+-- Implement a function that returns the sum of the last two digits of a number.
+--
+-- >>> sumLast2 42
+-- 6
+-- >>> sumLast2 134
+-- 7
+-- >>> sumLast2 1
+-- 1
+--
+-- Try to introduce variables in this task (either with let-in or where) to avoid
+-- specifying complex expressions.
+sumLast2 :: Int -> Int
+sumLast2 n = lastDigit n + beforeLast n
   where
-    square :: Double -> Double
-    square x = x ^ 2
-@
-
-You can define multiple functions inside __where__!
-Just remember to keep proper indentation.
--}
-
-{- |
-=⚔️= Task 9
-
-Implement a function that returns the sum of the last two digits of a number.
-
->>> sumLast2 42
-6
->>> sumLast2 134
-7
->>> sumLast2 1
-1
-
-Try to introduce variables in this task (either with let-in or where) to avoid
-specifying complex expressions.
--}
-
-sumLast2 n = error "sumLast2: Not implemented!"
-
-
-{- |
-=💣= Task 10*
-
-You did it! You've passed all the challenges in your first training!
-Congratulations!
-Now, are you ready for the boss at the end of this training???
-
-Implement a function that returns the first digit of a given number.
-
->>> firstDigit 230
-2
->>> firstDigit 5623
-5
-
-You need to use recursion in this task. Feel free to return to it later, if you
-aren't ready for this boss yet!
--}
-
-firstDigit n = error "firstDigit: Not implemented!"
-
+    beforeLast :: Int -> Int
+    beforeLast x = lastDigit (div ((abs x) - lastDigit x) 10)
```

Comment:
Great!
For `div` the last digit won't matter, so it is okay not to subtract it in here 🙂

Reply:
Thanks for your feedback @chshersh , I will try to solve the following chapters!!